### DISCUSSION TITLE
Use flog symbols for drawing commit trees

### DISCUSSION THREAD

**Discussion by jbergstroem** [jesseduffield/lazygit#4062] [Ideas]
[Flog symbols](https://github.com/rbong/flog-symbols) is a set of icons that a growing set of terminals (kitty, ghostty, wezterm wip) have decided to support via nerd-fonts. It would be visually pleasing to see lazygit use these as well!

Examples in use:
- https://github.com/SuperBo/fugit2.nvim
- https://github.com/isakbm/gitgraph.nvim
- https://github.com/rbong/vim-flog

**Comment by stefanhaller** [comment]
The screenshots in the example projects you linked to do indeed look awesome, but I get the impression that the symbols only work when you have blank lines between rows. If that's true, it's not an option for lazygit; we don't want to waste so much vertical space.

**Reply by jbergstroem** [reply]
> The screenshots in the example projects you linked to do indeed look awesome, but I get the impression that the symbols only work when you have blank lines between rows. If that's true, it's not an option for lazygit; we don't want to waste so much vertical space.

It seems that the screenshots just adds an extra line for a description? At least this one:

![](https://raw.githubusercontent.com/SuperBo/fugit2.nvim/assets/assets/main_view.png)

"Superbo" is part of git metadata.

**Reply by stefanhaller** [reply]
I don't care what these programs do or don't include in their lists of commits (other screenshots had a blank line instead of the metadata line). My point is that in this part of the graph: 
<img width="385" alt="image" src="https://github.com/user-attachments/assets/0a7e2232-18c1-4d89-818b-408130fcf265" />
you need three lines to draw the graph symbols, but we only have two available in lazygit.

**Reply by jbergstroem** [reply]
> you need three lines to draw the graph symbols, but we only have two available in lazygit.

Took a closer look by testing it out in my terminal and you're right; you do need three.

**Comment by jbergstroem** [comment]
Closing as resolved since lazy-git needs to fit glyphs in a 2 line-height drawing mode, not three.
